pkg/networkservice/utils/metadata/client.go

```diff
@@ -39,18 +39,19 @@ func NewClient() networkservice.NetworkServiceClient {
 func (m *metaDataClient) Request(ctx context.Context, request *networkservice.NetworkServiceRequest, opts ...grpc.CallOption) (*networkservice.Connection, error) {
 	connID := request.GetConnection().GetId()
 
-	ctx = store(ctx, connID, &m.Map)
-
-	conn, err := next.Client(ctx).Request(ctx, request, opts...)
+	conn, err := next.Client(ctx).Request(store(ctx, connID, &m.Map), request, opts...)
 	if err != nil {
-		del(ctx, connID, &m.Map)
+		m.Map.Delete(connID)
```

Comment:
How is the Close call going to access the metadata it needs to clean up resources?  Before it was guaranteed access to that data by del()... now it will not have it at all and we should expect to leak all resources...

Reply:
Close is currently making:
```
next.Server(ctx).Close(store(…), conn)
```
So it has access to metadata.
Actually the problem before was Close making:
```
next.Server(ctx).Close(del(…), conn)
```
And so on double Close there were no metadata.